src/fondant/data_io.py

```diff
@@ -189,7 +191,8 @@ def write_dataframe(self, dataframe: dd.DataFrame) -> None:
 
         with ProgressBar():
             logging.info("Writing data...")
-            dd.compute(*write_tasks)
+            # alternative implementation possible: futures = client.compute(...)
+            dd.compute(*write_tasks, scheduler=dask_client)
```

Comment:
Alternative implementation: 

```python
futures = dask_client.compute(write_tasks)
wait(futures)
...
```

Here, we still have to investigate how we are addressing the oom issue. See proposal in #406